src/frontend/screens/ManualGpsScreen/index.tsx

```diff
@@ -0,0 +1,231 @@
+          mocked: false,
+          coords: {
+            latitude: convertedData.coords?.lat,
+            longitude: convertedData.coords?.lon,
+          },
+        },
+        manualLocation: true,
+      });
+
+      navigation.pop();
+    }
+
+    navigation.setOptions({
+      headerRight: () => (
+        <IconButton onPress={handleSavePress}>
+          <SaveIcon inprogress={false} />
+        </IconButton>
+      ),
+    });
+  }, [navigation, convertedData, updateObservationPosition]);
+
+  const locationCoordinates =
+    observationValue &&
+    observationValue.lat !== undefined &&
+    observationValue.lon !== undefined
+      ? {
+          lat: observationValue.lat,
+          lon: observationValue.lon,
+        }
+      : undefined;
+
+  return (
+    <View>
+      {/* TODO: Set `behavior` to "padding" when iOS is supported */}
+      <KeyboardAvoidingView>
+        <ScrollView contentContainerStyle={styles.scrollContentContainer}>
+          <View style={styles.formatSelect}>
+            <Text style={styles.inputLabel}>
+              <FormattedMessage {...m.coordinateFormat} />
+            </Text>
+            <Select
+              containerStyles={styles.selectContainer}
+              onChange={value => {
+                if (typeof value === 'number' || !isCoordinateFormat(value)) {
+                  return;
+                }
+
+                setManualCoordinateEntryFormat(value);
+              }}
+              options={ENTRY_FORMAT_OPTIONS}
+              selectedValue={entryCoordinateFormat}
+            />
+          </View>
+
+          <View style={styles.formContainer}>
+            {entryCoordinateFormat === 'dd' ? (
+              <DdForm
+                initialCoordinates={locationCoordinates}
+                onValueUpdate={setConvertedData}
+              />
+            ) : entryCoordinateFormat === 'dms' ? (
+              <DmsForm
+                initialCoordinates={locationCoordinates}
+                onValueUpdate={setConvertedData}
+              />
+            ) : (
+              <UtmForm
+                initialCoordinates={locationCoordinates}
+                onValueUpdate={setConvertedData}
+              />
+            )}
+          </View>
+        </ScrollView>
+      </KeyboardAvoidingView>
+    </View>
+  );
+};
+
+export function createNavigationOptions({
+  intl,
+}: {
+  intl: (title: MessageDescriptor) => string;
+}) {
+  return (): NativeStackNavigationOptions => {
+    return {
+      headerTitle: intl(m.title),
+      headerRight: () => (
+        <IconButton>
+          <SaveIcon inprogress={false} />
+        </IconButton>
+      ),
+    };
+  };
+}
+
+function entryCoordinateFormatSelector(
+  state: Parameters<Parameters<typeof usePersistedSettings>[0]>[0],
+) {
+  return state.manualCoordinateEntryFormat;
+}
+
+function observationValueSelector(
+  state: Parameters<Parameters<typeof usePersistedDraftObservation>[0]>[0],
+) {
+  return state.value;
+}
```

Comment:
probably unnecessary but small easy-to-implement optimization when working with zustand selectors